### PR TITLE
Fix ideal loads zone air flow balance

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
Addresses #6382.

The ideal loads system was not always setting the correct mass flow rate on the exhaust node. When used together with an airloop serving the same zone this  caused the return air flow rate for the airloop to be incorrect. For the defect file which has a VAV system together with the ideal loads system, this resulted in the VAV system operating at nearly full flow any time the ideal loads system was operating.  
Always settting the mass flow rate of ideal loads exhaust node (or return node if there is no exahaust node) to match the ideal loads supply flow solves this issue.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - [x] If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [x] Performance: CI Linux results include performance check -- verify this
 - [x] Unit Test(s)

### CODE REVIEW

**Comment by nrel-bot-3** [comment]
@lgu1234 @lgentile it has been 14 days since this pull request was last updated.

File: src/EnergyPlus/PurchasedAirManager.cc [hunk 1]
```diff
@@ -2307,6 +2307,10 @@ namespace PurchasedAirManager {
 		if ( PurchAir( PurchAirNum ).PlenumExhaustAirNodeNum > 0 ) {
 			Node( PurchAir( PurchAirNum ).PlenumExhaustAirNodeNum ).MassFlowRate = SupplyMassFlowRate;
 			Node( RecircNodeNum ).MassFlowRate = SupplyMassFlowRate;
+		} else {
+			if ( PurchAir( PurchAirNum ).ZoneExhaustAirNodeNum > 0 && abs( Node( PurchAir( PurchAirNum ).ZoneExhaustAirNodeNum ).MassFlowRate - SupplyMassFlowRate) > rTinyValue ) {
+				Node( PurchAir( PurchAirNum ).ZoneExhaustAirNodeNum ).MassFlowRate = SupplyMassFlowRate;
+			}
```
Comment (mjwitte) [inline src/EnergyPlus/PurchasedAirManager.cc]: @lgu1234 I am not sure that this solution is general. If there is no exhaust node, the airflow imbalance will still occur. 

An better solution is to *always* set the recirc node mass flow rate equal to the supply flow rate.  The current line 2309
```
Node( RecircNodeNum ).MassFlowRate = SupplyMassFlowRate;
```
should just come out of the above if block and always execute.

Also, [this block](https://github.com/NREL/EnergyPlus/blob/017868e84f0eb0de7711a942c72aebc273bafcc8/src/EnergyPlus/PurchasedAirManager.cc#L2514-L2517) in `CalcPurchAirMixedAir` should be deleted.
a) This function shouldn't be touching any node massflowrates in the first place (or any node conditions at all)
b) The recirc node should always be set to `SupplyMassFlowRate`. Setting it to `RecircMassFlowRate` was just plain wrong.

Comment (lgu1234) [reply src/EnergyPlus/PurchasedAirManager.cc]: @mjwitte Here is my understanding. There are 3 nodes in the system, zone inlet node, zone exhaust node, and recirc node. 

If we set Node( RecircNodeNum ).MassFlowRate = SupplyMassFlowRate, and the ideal system is a DOAS system, do we have a relief node to handle outdoor air. If we don't have a relief node, here is my vision:
Node( RecircNodeNum ).MassFlowRate = 0.0;
Node( ZoneInletNodeNum ).MassFlowRate = SupplyMassFlowRate;
Node( ExhaustNodeNum ).MassFlowRate = SupplyMassFlowRate;

**Comment by mjwitte** [comment]
The exhaust node is optional.  If present then RecircNodeNum = ExhaustNodeNum. If not, then RecircNodeNum = ReturnNodeNum. So only two of these are ever active. There is an implied relief flow but no relief node.


-------- Original message --------

**Comment by mjwitte** [comment]
@lgu1234 This looks good now, as far as fixing the flow rate issue.  There's still something wrong with EMS actuating ideal loads - the warnings about supply humidity exceeding saturation should not be happening.  There were there before and after this fix.  A version of the defect file without any EMS code does not have these warnings but still exhibits the flow rate problem and the fix.  Will add more to #5788 which already identifies one problem with EMS and ideal loads.

**Pull request merged**